build/build.mk

```diff
@@ -87,23 +97,44 @@ define generatePlatformBuildTarget
 build-$(3)-$(1)-$(2):
   $(eval BINS_OUT_DIR_$(1)_$(2) := $(OUT_DIR)/$(1)_$(2)/$(BUILDTYPE_DIR))
 	@echo "$(ARROW) Building $(3) on $(1)/$(2) to $(BINS_OUT_DIR_$(1)_$(2))/$(3)$(BINARY_EXT)"
-	GOOS=$(1) GOARCH=$(2) go build \
+	cd $(4) && GOOS=$(1) GOARCH=$(2) go build \
 		-v \
 		-gcflags $(GCFLAGS) \
-		-ldflags=$(LDFLAGS) \
-		-o $(BINS_OUT_DIR_$(1)_$(2))/$(3)$(BINARY_EXT) \
-		$(4)/
+		-ldflags $(LDFLAGS) \
+		-o $(CURDIR)/$(BINS_OUT_DIR_$(1)_$(2))/$(3)$(BINARY_EXT)
 endef
 
 # defines a target for each binary
 GOOSES := darwin linux windows
 GOARCHES := amd64 arm arm64
-BINARIES := docgen rad applications-rp ucpd controller
-$(foreach ITEM,$(BINARIES),$(eval $(call generateBuildTarget,$(ITEM),./cmd/$(ITEM))))
-$(foreach ARCH,$(GOARCHES),$(foreach OS,$(GOOSES),$(foreach ITEM,$(BINARIES),$(eval $(call generatePlatformBuildTarget,$(OS),$(ARCH),$(ITEM),./cmd/$(ITEM))))))
 
-# list of 'outputs' to build for all binaries
-BINARY_TARGETS:=$(foreach ITEM,$(BINARIES),build-$(ITEM))
+# List of binaries to build.
+# Format: binaryName:binaryMainDirectory
+# Example: docgen:./cmd/docgen
+# Dockerfile should be present in the binaryMainDirectory.
```

Comment:
This doesn't seem right. We don't keep the dockerfile right next to the `main.go`.

Reply:
Removed this one. My bad.